scholarx/archive/functions.js

```diff
@@ -52,13 +58,128 @@ async function getData() {
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfilesData });
+    $("#mentorProfiles").html(mentorProfiles);
 
-    const menteeProfiles = payload.mentee2021.concat(payload.mentee2020).concat(payload.mentee2019);
-    let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfiles });
-    $("#menteeProfiles").html(mentee_Profiles);
+    menteeProfilesData = mentee[2019].concat(mentee[2020]).concat(mentee[2021]);
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfilesData });
+    $("#menteeProfiles").html(menteeProfiles);
 }
 loadData();
+function renderProfiles(mentorYear,menteeYear) {
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorYear });
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeYear });
+    $("#mentorProfiles").html(mentorProfiles);
+    $("#menteeProfiles").html(menteeProfiles);
+}
+function renderAllProfiles() {
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfilesData });
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfilesData });
+    $("#mentorProfiles").html(mentorProfiles);
+    $("#menteeProfiles").html(menteeProfiles);
+}
+//function to return true if all industry checkboxes are unchecked.
+function allFalse(){
+    let number = 0;
+    for(let i=1; i<7; i++){
+        if(document.getElementById("industry_"+i).checked == false){
+            number++;
+        }
+    }
+    if(number==6){
+        return true
+    }else{
+        return false
+    }
+}
+//function to uncheck checkboxes
+function uncheckIndustryFilters(){
+    for(let i=1; i<7; i++){
+        document.getElementById("industry_"+i).checked = false;
+    }
+}
+function uncheckYearByFilters(){
+    let year = 2019
+    for(let i=0; i<4; i++){
+        document.getElementById(year).checked = false
+        year++
+    }
+}
+//year by filters
+function filterByYear(){
+    uncheckIndustryFilters();
+    let year = 2019;
+    let mentorsData = [];
+    let menteesData = [];
+    if(document.getElementById("2019").checked == false && document.getElementById("2020").checked == false && document.getElementById("2021").checked == false){
+        renderAllProfiles();
+    }else{
+        for(let i=0; i<3; i++){
+            if(document.getElementById(year).checked)
+            {
+                mentorsData = mentorsData.concat(mentor[year]),
+                dataHolder = mentorsData,
+                menteesData = menteesData.concat(mentee[year]);
+                renderProfiles(mentorsData,menteesData)
+            }
+            year++;
+        }
+    }
+}
+//filter by industry
+function filterProfiles(value){
+    let mentorByIndustry = {
```

Comment:
According to your approach, we will have to call the `uncheckYearByFilters()` function inside the `filterByIndustry` function. 
(It will automatically clear the industry filters if we select a year filter and vice versa)